packages/runtime/container-runtime/src/runningSummarizer.ts

```diff
@@ -68,12 +71,34 @@ export class RunningSummarizer implements IDisposable {
             summaryCollection,
             cancellationToken,
             stopSummarizerCallback,
-            options);
+            options,
+            runtime);
 
         await summarizer.waitStart();
 
-        // Run the heuristics after starting
+        // Update heuristic counts
+        // By the time we get here, there are potentially ops missing from the heuristic summary counts
+        // Examples of where this could happen:
+        // 1. Op is processed during the time that we are initiating the RunningSummarizer instance but before we
+        //    listen for the op events (will get missed by the handlers in the current workflow)
+        // 2. Op was sequenced after the last time we summarized (op sequence number > summarize ref sequence number)
+        const diff = runtime.deltaManager.lastSequenceNumber - (
+            heuristicData.lastSuccessfulSummary.refSequenceNumber
+            + heuristicData.numSystemOps
+            + heuristicData.numNonSystemOps);
+        if (diff > 0) {
+            // Split the diff 50-50 and increment the counts appropriately
+            heuristicData.numSystemOps += Math.ceil(diff / 2);
+            heuristicData.numNonSystemOps += Math.floor(diff / 2);
```

Comment:
minor - off-by-1 error due to rounding of odd/2 calculus

Reply:
Wouldn't `Math.ceil` and `Math.floor` catch this case? From what I remember, Javascript automatically includes the decimals since there is only the `number` type.

Reply:
looks good in terms of math.